integration-tests/src/test_loop/tests/fix_stake_threshold.rs

```diff
@@ -0,0 +1,131 @@
+use crate::test_loop::builder::TestLoopBuilder;
+use crate::test_loop::env::TestLoopEnv;
+use crate::test_loop::utils::validators::get_epoch_all_validators;
+use crate::test_loop::utils::ONE_NEAR;
+use itertools::Itertools;
+use near_async::test_loop::data::TestLoopData;
+use near_async::time::Duration;
+use near_chain_configs::test_genesis::TestGenesisBuilder;
+use near_o11y::testonly::init_test_logger;
+use near_primitives::epoch_manager::EpochConfigStore;
+use near_primitives::num_rational::Rational32;
+use near_primitives::test_utils::create_test_signer;
+use near_primitives::types::AccountId;
+use near_primitives::types::AccountInfo;
+use near_primitives_core::version::ProtocolFeature;
+
+const NUM_SHARDS: u128 = 6;
```

Comment:
nit: number of mainnet shards changes with time.
Could you consider using test epoch config store where number of shards is fixed? The API is not straightforward but there should be examples around.